components/support/crypto/nss-sys/src/backends/mod.rs

```diff
@@ -0,0 +1,8 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#[cfg(not(ios))]
+pub(crate) mod dylib_nss;
+#[cfg(ios)]
+pub(crate) mod geckoview_nss;
```

Comment:
Perhaps I don't understand how `#[cfg(ios)]` works, but do me this says "when not on ios, use dylib_nss, which links against libnss; when on ios use geckoview_nss which loads it via dlopen". But that sounds like the opposite of what you described in the PR comment, where we want to use dlopen on android. What am I missing?

Reply:
Sorry that's an mistake on my part. I'm actually very confused because XCode was complaining about not being able to link with some NSS symbols... despite in theory using the `dlopen` version! My hunch is that `#[cfg(ios)]` is actually incorrect and we should use `#[cfg(target_os = "ios")]` instead.

Reply:
Yup:
```
rustc --print cfg --target aarch64-apple-ios
debug_assertions
target_arch="aarch64"
target_endian="little"
target_env=""
target_family="unix"
target_os="ios"
target_pointer_width="64"
target_vendor="apple"
unix
```
TIL

Reply:
Yes, AFAIK #[cfg(ios)] is meaningless, I'm surprised it's not an error. Confusingly, `#[cfg(unix)]` and `#[cfg(windows)]` exist, but are shorthand for `#[cfg(target_family = "unix")]` and `#[cfg(target_family = "windows")]` respectively, but this does not generalize.

https://doc.rust-lang.org/reference/conditional-compilation.html is a full set of documentation.